straxen/plugins/bayes_peaklet_classification.py

```diff
@@ -0,0 +1,138 @@
+                                                   waveforms, quantiles)
+
+        return dict(time=peaks['time'],
+                    endtime=peaks['time'] + peaks['dt'] * peaks['length'],
+                    ln_prob_s1=ln_prob_s1,
+                    ln_prob_s2=ln_prob_s2
+                    )
+
+def compute_wf_and_quantiles(peaks: np.ndarray, bayes_n_nodes: int):
+    """
+    Compute waveforms and quantiles for a given number of nodes(atributes)
+    :param peaks:
+    :param bayes_n_nodes: number of nodes or atributes
+    :return: waveforms and quantiles
+    """
+    waveforms = np.zeros((len(peaks), bayes_n_nodes))
+    quantiles = np.zeros((len(peaks), bayes_n_nodes))
+
+    num_samples = peaks['data'].shape[1]
+    step_size = int(num_samples/bayes_n_nodes)
+    steps = np.arange(0, num_samples+1, step_size)
+
+    data = peaks['data'].copy()
+    dts = peaks['dt'].copy()
+    data[data < 0.0] = 0.0
+    for i, p in enumerate(peaks):
+        sample_number = np.arange(0, num_samples+1, 1)*p['dt']
+        frac_of_cumsum = np.append([0.0], np.cumsum(data[i, :]) / np.sum(data[i, :]))
+        cumsum_steps = np.interp(np.linspace(0., 1., bayes_n_nodes, endpoint=False), frac_of_cumsum, sample_number)
+        cumsum_steps = np.append(cumsum_steps, sample_number[-1])
+        quantiles[i, :] = cumsum_steps[1:] - cumsum_steps[:-1]
+
+    for j in range(bayes_n_nodes):
+        waveforms[:, j] = np.sum(data[:, steps[j]:steps[j+1]], axis=1)
+    waveforms = waveforms/(dts*step_size)[:, np.newaxis]
+
+    del data
+    return waveforms, quantiles
+
+def compute_inference(bins: int, bayes_n_nodes: int, cpt: np.ndarray, n_bayes_classes: int, class_prior: np.ndarray,
+                      waveforms: np.ndarray, quantiles: np.ndarray):
+    """
+    Bin the waveforms and quantiles according to Bayes bins and compute inference
+    :param bins: Bayes bins
+    :param bayes_n_nodes: number of nodes or atributes
+    :param cpt: conditioanl probability tables
+    :param n_bayes_classes: number of classes
+    :param class_prior: class_prior
+    :param waveforms: waveforms
+    :param quantiles: quantiles
+    :return: ln probability per class S1/S2
+    """
+    # Bin the waveforms and quantiles.
+    waveform_bin_edges = bins[0, :][bins[0, :] > -1]
+    waveform_num_bin_edges = len(waveform_bin_edges)
+    quantile_bin_edges = bins[1, :][bins[1, :] > -1]
+    quantile_num_bin_edges = len(quantile_bin_edges)
+
+    waveform_values = np.digitize(waveforms, bins=waveform_bin_edges)-1
+    waveform_values[waveform_values < 0] = int(0)
+    waveform_values[waveform_values > int(waveform_num_bin_edges - 2)] = int(waveform_num_bin_edges - 2)
+
+    quantile_values = np.digitize(quantiles, bins=quantile_bin_edges)-1
+    quantile_values[quantile_values < 0] = int(0)
+    quantile_values[quantile_values > int(quantile_num_bin_edges - 2)] = int(quantile_num_bin_edges - 2)
```

Comment:
Can you explain what you are doing here I cannot follow. Also having the suffix `_values` is a bit miss leading. If I understand np.digitize correctly it will return indices not the waveform values.